godot-core/src/registry.rs

```diff
@@ -418,9 +570,22 @@ pub mod callbacks {
         string.move_string_ptr(out_string);
     }
 
+    #[cfg(before_api = "4.2")]
+    pub unsafe extern "C" fn on_notification<T: cap::GodotNotification>(
+        instance: sys::GDExtensionClassInstancePtr,
+        what: i32,
+    ) {
+        let storage = as_storage::<T>(instance);
+        let mut instance = storage.get_mut();
+
+        T::__godot_notification(&mut *instance, what);
+    }
+
+    #[cfg(since_api = "4.2")]
     pub unsafe extern "C" fn on_notification<T: cap::GodotNotification>(
         instance: sys::GDExtensionClassInstancePtr,
         what: i32,
+        _reversed: sys::GDExtensionBool,
```

Comment:
No need to change things now, but do you think the user could in the future benefit from having this `reversed` flag available inside `on_notification`?